src/Observable.php

```diff
@@ -408,7 +478,7 @@ public function mapTo($value)
      * @operator
      * @reactivex map
      */
-    public function select(callable $selector)
+    public function select(callable $selector): AnonymousObservable
```

Comment:
I think changing this to `Observable` makes more sense. It would introduce a break in the API if the internal details of how the operator works are changed. (probably less likely on this particular operator)

Returning the more restrictive `AnonymousObservable` does not gain you anything.